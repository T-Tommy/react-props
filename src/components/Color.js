import React from 'react';
import PropTypes from 'prop-types';
import toHexColor from '../services/rgb-converter';


function Color({ name, rgb }) {
  const { r, g, b } = rgb;
  const dlStyle = {
    color: toHexColor(rgb)
  };

  return (
    <dl style={dlStyle}>
      <dt>Name</dt>
      <dd>{ name }</dd>

      <dt>Hex</dt>
      <dd>{ toHexColor(rgb) }</dd>

      <dt>RGB</dt>
      <dd>Red - { r }</dd>
      <dd>Green - { g }</dd>
      <dd>Blue - { b }</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string,
  rgb: PropTypes.object.isRequired
};

export default Color;
