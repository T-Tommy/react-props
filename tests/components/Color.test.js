import React from 'react';
import { shallow } from 'enzyme';
import Color from '../../src/components/Color';

describe('Color component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color name="blue" rgb={ { r: 0, g: 0, b: 255 } }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
