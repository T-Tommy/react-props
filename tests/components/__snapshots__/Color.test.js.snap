// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a Color 1`] = `
<dl
  style={
    Object {
      "color": "#0000FF",
    }
  }
>
  <dt>
    Name
  </dt>
  <dd>
    blue
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #0000FF
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    Red - 
    0
  </dd>
  <dd>
    Green - 
    0
  </dd>
  <dd>
    Blue - 
    255
  </dd>
</dl>
`;
